Proteomics/RetentionTimePrediction/SSRCalc3.cs

```diff
@@ -0,0 +1,1486 @@
+               double rcs, double rc1s, double rc2s, double rcns, double rcn2s,
+               double undkrh, double amass,
+               double ct, double nt, double pk,
+               double h2bascore, double h2cmult
+            )
+            {
+                RC = rc;
+                RC1 = rc1;
+                RC2 = rc2;
+                RCN = rcn;
+                RCN2 = rcn2;
+                RCS = rcs;
+                RC1S = rc1s;
+                RC2S = rc2s;
+                RCNS = rcns;
+                RCN2S = rcn2s;
+                UndKRH = undkrh;
+                AMASS = amass;
+                CT = ct;
+                NT = nt;
+                PK = pk;
+                H2BASCORE = h2bascore;
+                H2CMULT = h2cmult;
+            }
+        }
+        /*
+         * Translator2 note: The code for the Isoparams array was found in
+         *      the Java version, but never used.  Refering to the Perl
+         *      version showed that the only place these values were used
+         *      was in the electric_scale() function, which in turn was never
+         *      used.  Both the array and function are included here for
+         *      completeness, but commented out, since they are never used.
+         *
+        private class Isoparams
+        {
+            public double emin { get; private set; }
+            public double emax { get; private set; }
+            public double eK { get; private set; }
+
+            public Isoparams(double EMIN, double EMAX, double EK)
+            {
+                emin = EMIN; emax = EMAX; eK = EK;
+            }
+        }
+
+        private static readonly Isoparams[] ISOPARAMS = new[]
+        {
+            new Isoparams(3.8, 4.0, 0.880),
+            new Isoparams(4.0, 4.2, 0.900),
+            new Isoparams(4.2, 4.4, 0.920),
+            new Isoparams(4.4, 4.6, 0.940),
+            new Isoparams(4.6, 4.8, 0.960),
+            new Isoparams(4.8, 5.0, 0.980),
+            new Isoparams(5.0, 6.0, 0.990),
+            new Isoparams(6.0, 7.0, 0.995),
+            new Isoparams(7.0, 8.0, 1.005),
+            new Isoparams(8.0, 9.0, 1.010),
+            new Isoparams(9.0, 9.2, 1.020),
+            new Isoparams(9.2, 9.4, 1.030),
+            new Isoparams(9.4, 9.6, 1.040),
+            new Isoparams(9.6, 9.8, 1.060),
+            new Isoparams(9.8, 10.0, 1.080)
+        };
+
+        // convert electric to scaler - v 2,3 algorithms
+        private static double electric_scale(double v)
+        {
+            double best=1.0;
+
+            // Translator2 Note: this is commented out in the perl source
+            // if (NOELECTRIC==1) { return 1.0; }
+
+            foreach (Isoparams p in ISOPARAMS)
+            {
+                if (v > p.emin && v < p.emax)
+                    best= p.eK;
+            }
+
+            return best;            
+        }
+        */
+    }
+
+    internal static class HelpersLocal
```

Comment:
This is a really long class, with nested classes and such. Usually, that points to possibly refactoring for clarity.

Reply:
I agree, but I'm not going to monkey with the code more than I have to if it works. It's been a point of mine to maintain as much of the original code as possible, so I don't feel good about refactoring it right now, especially if we have plans to implement better HPLC prediction in the future. I view this as a temporary crutch and a somewhat historic implementation for people who might want to compare with the original SSRCalc algorithm.

Reply:
Sounds good.